### ISSUE TITLE
Lazy Loading of charts not working when height of container of charts is set

### ISSUE THREAD

**Issue by sarthak1995** [amcharts/amcharts4#2316]
Hi There,

I have multiple containers in a React Page, with each container having multiple rows, and each row has multiple charts in it. The Page is loaded with around 50 charts at a time which freezes the whole browser window. In your documentation, I came across the lazy loading concept but it seems to work only when height of the charts container is not set. If set, only charts visible on initial view loads, and the remaining doesn't load on scrolling. 

Hoping to hear from you on this issue.

**Comment by martynasma** [comment]
I don't think setting height on a chart container may affect its appearance when lazy-loading is active.

Can you provide an example of such behavior which we can run?

**Comment by sarthak1995** [comment]
Hi,

https://codepen.io/95sarthak/pen/abvzdLV

I hope it helps.

**Comment by martynasma** [comment]
Thanks. It seems you have charts in the scrollable container, whereas we watch the window for scroll events by default.

So in your case you just need to specify which is container for charts using global option `viewportTarget`:

```
am4core.options.viewportTarget = document.getElementById("charts");
```

**Comment by sarthak1995** [comment]
Thanks @martynasma 

`am4core.options.viewportTarget` seems to work.

What if I have multiple chart containers where I am reusing the same chart files to make charts?

**Comment by martynasma** [comment]
Do you? That seems like a very odd setup. Not very good UX, having to manipulate multiple scrollbars.

Answering your question, that would involve much more complex solution, e.g. setting up your own scroll even handlers, and triggering amCharts-related API calls.

**Comment by sarthak1995** [comment]
@martynasma, Consider multiple scrollable tables with charts in the cell of the tables. So right now, all the charts load even if the table cells are not visible to the user. Rows are not much but each row contains 2-3 charts(mini)

So I think if we can have lazy loading so that the chart loads the user sees it.

**Comment by martynasma** [comment]
Good point. Give us a moment to think about it. Maybe we can make `viewportTarget` accept an array of references instead of one.

Will let you know when/if that gets implemented.

**Comment by sarthak1995** [comment]
Yes sure. @martynasma

**Comment by martynasma** [comment]
This will be in the next version we released. At some point next week.

Will notify you here when the new version is out.

**Comment by martynasma** [comment]
Fixed in [4.9.14](https://github.com/amcharts/amcharts4/releases/tag/4.9.14).

## [4.9.14] - 2020-04-13

### Added
- Global options setting added `am4core.options.animationsEnabled` (default `true`). Set it to `false` to **disable all animations on all charts**. Can be used to toggle animations on and off dynamically.
- `SliceGrouper` plugin: new setting `limit` added. If set to a number, will ignore `threshold` but will show only first X slices and will group the rest.
- New PDF format option for `Export`: `align`. Available options: `"left"` (default), `"center"`, and `"right"`.
- New `ValueAxis` property: `adjustLabelPrecision` (default: `true`). `true` means that **all** labels on `ValueAxis` (except zero) will maintain the same number of decimals, e.g. `1.0`, `1.5`, `2.0` versus `1`, `1.5`, `2`. Enabling it will ignore number of active decimals set via `numberFormat`.
- `DateFormatter` is now exposing list of various named time units: `months`, `monthsShort`, `weekdays`, and `weekdaysShort`. You can now resolve numeric value to an English name (e.g. `chart.dateFormatter.weekdaysShort[2]`) or to your active locale (`chart.dateFormatter.language.translate(chart.dateFormatter.weekdaysShort[1])`).

### Changed
- New property `adjustLabelPrecision` hsa a default of `true`, which means that all labels will now have same number of decimals on `ValueAxis`. Set `adjustLabelPrecision = false`, to disable and revert behavior to how it was before this update (`1`, `1.5`, `2`).
- `am4core.options.viewportTarget` now accepts array of references if you have multiple scrollable containers holding charts and have `onlyShowOnViewport` enabled.
- Default `DateAxis.periodChangeDateFormats` for `"hour"` changed from `"_date_hour"` to `"_date_day"` so that the format would differ for midnight (hour 00:00).

### Fixed
- Parsing short month names (`"MMM"`) from string-based dates was failing on May.
- When exporting chart to PDF, on narrow charts, sometimes the whole chart would go to the next page.
- Having `null` values in data was causing `maxTooltipDistance == -1` to show multiple tooltips instead of one.
- It was impossible to have individual `innerRadius` for `AxisRendererRadial` and `AxisRendererCircular` - the `RadarChart`'s `innerRadius` was overriding them.
- Removing axis ranges from an axis could result in a JS error.
- In some cases, when `DateAxis` properties `startLocation` and `endLocation` were not defaults (0 and 1), the chart might flicker from one grouped data period to another continously. It was only happening with some particular `groupCount` values and data item counts, though.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
